blocks/monitoring/aggregation.py

```diff
@@ -138,11 +138,62 @@ def get_aggregator(self):
         return aggregator
 
 
-def mean(numerator, denominator=1.):
+def mean(numerator, denominator=1., name=None):
     """Mean of quantity (numerator) over a number (denominator) values."""
     variable = numerator / denominator
     variable.tag.aggregation_scheme = Mean(numerator, denominator)
-    variable.name = numerator.name
+    if name is not None:
```

Comment:
We don't really have `extension.monitor(...)`  in Blocks now. One creates `TrainingDataMonitoring` or `DataStreamMonitoring` with the list of all variables to be monitored. 

There is also a homogeneity argument: if you create a variable for monitoring without adding an aggregation scheme, you have to give it a name in a separate line. 

Maybe we should make sure that `named_copy` doesn't lose the aggregation scheme and whatever else it loses?